src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String EXCEPTION_NULL = "Cannot be null";
+    private static final String INDEX_OUT_OF_BOUND = "Index  is out of bounds: ";
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (elementData.length == size) {
+            grow();
+        }
+        for (int i = size - 1; i >= index; i--) {
+            elementData[i + 1] = elementData[i];
+        }
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new NullPointerException(EXCEPTION_NULL);
```

Comment:
bad practice to throw NullPointerException, we need to avoid this situation in code